services/QuillLMS/engines/evidence/lib/evidence/evidence/profanity.rb

```diff
@@ -0,0 +1,26 @@
+module Evidence 
+  class Profanity 
+    def self.profane?(word)
```

Comment:
This might be a premature optimization (this may be super fast as is), but do we need to check for profanity in the entry word by word? With 450 "bad words", if there are 50 words in an entry, that's 50*450 or 22,500 comparisons. vs. 450 comparisons if you just checked for profane words in the entire entry at once.

Could we do something like this to trim everything that isn't in the entry and do the deeper inspection ONLY on items in the entry.

```ruby
words_in_entry = bad_words.gsub('*','').select {|word| entry.include?(word) } 
# then deeper check on words_in_entry

Reply:
@happythenewsad I don't believe this one was addressed. What are your thoughts on this? Most students won't write profanity, so the most common run case will be running through all of the words.

Reply:
@dandrabik Sorry for the confusion. I self-commented a reply (above), but it was displaying only for me until I hit "Start a review" just now. Hopefully you see it now. But to reiterate, I think we should ship it as is, or do a once-and-for-all optimization (Bloom filter).